pkg/agent/openflow/pod_connectivity.go

```diff
@@ -0,0 +1,134 @@
+	cookieAllocator cookie.Allocator,
+	ipProtocols []binding.Protocol,
+	nodeConfig *config.NodeConfig,
+	networkConfig *config.NetworkConfig,
+	connectUplinkToBridge bool,
+	enableMulticast bool) *featurePodConnectivity {
+	ctZones := make(map[binding.Protocol]int)
+	gatewayIPs := make(map[binding.Protocol]net.IP)
+	localCIDRs := make(map[binding.Protocol]net.IPNet)
+	nodeIPs := make(map[binding.Protocol]net.IP)
+	for _, ipProtocol := range ipProtocols {
+		if ipProtocol == binding.ProtocolIP {
+			ctZones[ipProtocol] = CtZone
+			gatewayIPs[ipProtocol] = nodeConfig.GatewayConfig.IPv4
+			localCIDRs[ipProtocol] = *nodeConfig.PodIPv4CIDR
+			nodeIPs[ipProtocol] = nodeConfig.NodeIPv4Addr.IP
+		} else if ipProtocol == binding.ProtocolIPv6 {
+			ctZones[ipProtocol] = CtZoneV6
+			gatewayIPs[ipProtocol] = nodeConfig.GatewayConfig.IPv6
+			localCIDRs[ipProtocol] = *nodeConfig.PodIPv6CIDR
+			nodeIPs[ipProtocol] = nodeConfig.NodeIPv6Addr.IP
+		}
+	}
+
+	return &featurePodConnectivity{
+		cookieAllocator:       cookieAllocator,
+		ipProtocols:           ipProtocols,
+		nodeFlowCache:         newFlowCategoryCache(),
+		podFlowCache:          newFlowCategoryCache(),
+		gatewayIPs:            gatewayIPs,
+		ctZones:               ctZones,
+		localCIDRs:            localCIDRs,
+		nodeIPs:               nodeIPs,
+		nodeConfig:            nodeConfig,
+		networkConfig:         networkConfig,
+		connectUplinkToBridge: connectUplinkToBridge,
+		enableMulticast:       enableMulticast,
+	}
+}
+
+func (f *featurePodConnectivity) initFlows(category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+
+	for _, ipProtocol := range f.ipProtocols {
+		if ipProtocol == binding.ProtocolIPv6 {
+			flows = append(flows, f.ipv6Flows(category)...)
+		} else if ipProtocol == binding.ProtocolIP {
+			flows = append(flows, f.arpNormalFlow(category))
+			flows = append(flows, f.arpResponderFlow(category, f.nodeConfig.GatewayConfig.IPv4, f.nodeConfig.GatewayConfig.MAC))
+		}
+	}
+	flows = append(flows, f.defaultDropFlows(category)...)
+	flows = append(flows, f.l3FwdFlowToLocalCIDR(category)...)
+	flows = append(flows, f.l3FwdFlowToNode(category)...)
+	flows = append(flows, f.l3FwdFlowToExternal(category))
+	flows = append(flows, f.decTTLFlows(category)...)
+	flows = append(flows, f.conntrackFlows(category)...)
+	flows = append(flows, f.l2ForwardOutputFlow(category))
+	if f.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
+		flows = append(flows, f.l3FwdFlowRouteToGW(category)...)
+		// If IPv6 is enabled, this flow will never get hit.
+		// Replies any ARP request with the same global virtual MAC.
+		flows = append(flows, f.arpResponderStaticFlow(category))
+	}
+	return flows
+}
+
+func (f *featurePodConnectivity) replayFlows() []binding.Flow {
+	var flows []binding.Flow
+
+	// Get fixed flows.
+	for _, defaultFlows := range [][]binding.Flow{f.defaultTunnelFlows, f.gatewayFlows, f.hostNetworkingFlows} {
+		for _, flow := range defaultFlows {
```

Comment:
Is it possible to create a new cache for these static flows?

Reply:
Sure. Added a new cache call `fixedFlows` to cache these static flows.